controllers/sms-gateway.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * This module implements GET and POST to support medic-gateway's API
+ * @see https://github.com/medic/medic-gateway
+ */
+
+var _ = require('underscore'),
+    db = require('../db'),
+    http = require('http'),
+    querystring = require('querystring'),
+    utils = require('./utils'),
+    messageUtils = require('./messages');
+
+require('lie/polyfill');
+
+var UNUSED = null; // this var is declared in various functions in the messages controller but never actually used
+
+function readBody(stream) {
+  var body = '';
+  return new Promise(function(resolve, reject) {
+    stream.on('data', function(data) {
+      body += data.toString();
+    });
+    stream.on('end', function() {
+      resolve(body);
+    });
+    stream.on('error', reject);
+  });
+}
+
+function readResBody(res) {
+  return readBody(res)
+    .then(function(body) {
+      if (res.statusCode >= 400) {
+        throw new Error('Bad status received: ' + res.statusCode + ': ' + body);
+      }
+      return body;
+    });
+}
+
+function saveToDb(gatewayRequest, wtMessage) {
+  var messageBody = querystring.stringify({
+    from: wtMessage.from,
+    message: wtMessage.content,
+    'medic-gateway_id': wtMessage.id,
```

Comment:
Because the object also includes:
```{
  "type": "message",
  "_id": ...
}```
so adding a property `message_id` doesn't seem like it would be very clear...